src/Components/Base/Badge/index.tsx

```diff
@@ -0,0 +1,48 @@
+import { ForwardedRef, forwardRef } from 'react';
+import { useTheme } from 'styled-components';
+import StyledBadge from './style';
+import { BadgeProp } from './type';
+
+/**
+ * Badge 컴포넌트를 사용하는 상위 요소의 position: relative 속성을 확인해주세요
+ */
+const Badge = forwardRef(
+  (
+    {
+      children,
+      size,
+      position,
+      backgroundColor,
+      textColor,
+      textSize,
+      ...props
+    }: BadgeProp,
+    ref: ForwardedRef<HTMLSpanElement>,
+  ) => {
+    const theme = useTheme();
+
+    const isSingleDigit =
+      String(children).length <= 2 || children === undefined;
+    const shape = isSingleDigit ? 'circle' : 'ellipse';
+    const isOverDigit = Number(children) >= 99;
+
+    return (
+      <StyledBadge
+        ref={ref}
+        $size={size || '0.8rem'}
+        $position={position || 'rightTop'}
+        $backgroundColor={backgroundColor || theme.colors.alert}
+        $textColor={textColor || theme.colors.buttonText}
+        $textSize={textSize || '0.4rem'}
```

Comment:
필수 x 선택 o size를 number로 받는다는 가정 o

size와 textSize가 둘 다 선택 속성이라서, size만 설정해주었을 때 textSize는 그대로 0.4rem으로 있게되는데 이걸 비율로 고쳐주면 조금 더 편리하게 사용할 수 있을 것 같습니다.

```suggestion
        $textSize={textSize || size / 2}
```

size / 2 비율로 넣어줬을 때 글씨 크기가 가장 무난해서 예시로 넣어보았습니다.
이건 개인적인 의견이니까 패스 하셔도 됩니다 ㅋㅋㅋ

Reply:
좋은 것 같아요! 그럼 size의 기본값을 prop 선언할 때 지정해줘야겠네요 👍🏻